File: summer/WebContent/js/system/collections/EmptyEnumerable.js
/**
 * EmptyEnumerable
 */

define(["dojo/_base/declare", "system/Type", "collections/IEnumerable", "internal.controls/EmptyEnumerator"], 
		function(declare, Type, IEnumerable, EmptyEnumerator){
	var EmptyEnumerable = declare("EmptyEnumerable", IEnumerable,{
//        IEnumerator IEnumerable.
        GetEnumerator:function()
        {
            return EmptyEnumerator.Instance;
        } 
	});
	
	Object.defineProperties(EmptyEnumerable,{
//        public static IEnumerable 
        Instance:
        {
            get:function()
            {
                if (EmptyEnumerable._instance == null) 
                {
                	EmptyEnumerable._instance = new EmptyEnumerable(); 
                } 
                return EmptyEnumerable._instance;
            } 
        }
	});
	
	EmptyEnumerable.Type = new Type("EmptyEnumerable", EmptyEnumerable, [IEnumerable.Type]);
	return EmptyEnumerable;
});

File: org.summer.view.widget/src/org/summer/view/widget/controls/internal/EmptyEnumerable.java
package org.summer.view.widget.controls.internal;

import org.summer.view.widget.collection.IEnumerable;
import org.summer.view.widget.collection.IEnumerator;

/// <summary> 
    /// Returns an Enumerable that is empty.
    /// </summary>
    /*internal*/public class EmptyEnumerable implements IEnumerable
    { 
        // singleton class, private ctor
        private EmptyEnumerable() 
        { 
        }
 
        public IEnumerator /*IEnumerable.*/GetEnumerator()
        {
            return EmptyEnumerator.Instance;
        } 

        /// <summary> 
        /// Read-Only instance of an Empty Enumerable. 
        /// </summary>
        public static IEnumerable Instance 
        {
            get
            {
                if (_instance == null) 
                {
                    _instance = new EmptyEnumerable(); 
                } 
                return _instance;
            } 
        }

        private static IEnumerable _instance;
    }